edge-util/src/Microsoft.Azure.Devices.Edge.Util/DefaultDictionary.cs

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Util
+{
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+    using System.Text;
+
+    /// <summary>
+    /// This acts exactly like a regular dictionary except that indexing [] a nonexistant key will instead generate a default value using the provided function.
+    /// </summary>
+    /// <typeparam name="TKey">Key type</typeparam>
+    /// <typeparam name="TValue">Value type</typeparam>
+    public class DefaultDictionary<TKey, TValue> : IDictionary<TKey, TValue>
```

Comment:
This is a LOT of boilerplate to get one behavior...and I think that behavior ("indexing [] a nonexistent key will instead generate a default value") gets used in exactly one place. Would a single helper function near the one call site be better? Otherwise I'd expect to see tests for `DefaultDictionary`.

Reply:
I mean, its 95% auto generated, and its a very simple pattern, just a wrapper for the dictionary. It is only used once rn, but its a useful class imo. I almost added it a while ago in a different pr but didn't end up needing it.

Reply:
Ok sure. Then it probably needs some basic tests, at least for the special behavior.